src/test/analyserwaveformtest.cpp

```diff
@@ -45,6 +45,49 @@ class AnalyzerWaveformTest : public MixxxTest {
         }
     }
 
+    void assertWaveformReference(
+            ConstWaveformPointer pWaveform,
+            const QString& reference_title) {
+        pWaveform->dump();
+
+        QFile f(QDir::currentPath() + "/src/test/reference_buffers/" + reference_title);
+        bool pass = true;
+        // If the file is not there, we will fail and write out the .actual
+        // reference file.
+        QByteArray actual = pWaveform->toByteArray();
+        QByteArray reference;
```

Comment:
Do we really want a bitwise comparision? Wouldn't it make sense to compare the waveformdata on the basis of the actual content? Not the memory or on-disk representation (which is here assumed to be the same too).

Reply:
This was the easiest way to have two independent paths for comparing the disc representation. 
With some efforts, we can also do it on the member variables, but that is something I do not want to spend